<img src="https://www.dmproperties.com/cms/wp-content/uploads/2015/10/crecimiento-sector-inmobiliario-espanol-610x407.jpg" alt="spotify" width="300"/>

<h1 style="text-align: center;">Limpieza de datos inmoviliarios Remax</h1>

## Librerias a usar

In [2]:
import pandas as pd
import matplotlib.pylab as plt

## Lectura del archivo CSV obtenido del webscrapping

In [3]:
file='C:\\Users\\ADMIN\\Desktop\\prueba1\\Remax_Venta.csv'

In [4]:
df = pd.read_csv(file)

In [5]:
df.head(10)

Tipo de Inmueble                           Direccion  \
0    DEPARTAMENTO FLAT EN ALQUILER              Lima, Lima, San Miguel   
1             CASA URBANA EN VENTA  Lima, Lima, San Juan de Lurigancho   
2  DEPARTAMENTO DUPLEX EN ALQUILER              Lima, Lima, Miraflores   
3             CASA URBANA EN VENTA             Lima, Lima, La Victoria   
4      TERRENO INDUSTRIAL EN VENTA                 Piura, Paita, Colan   
5              ALMACÉN EN ALQUILER                   Lima, Lima, Lince   
6     DEPARTAMENTO DUPLEX EN VENTA              Lima, Lima, Miraflores   
7       DEPARTAMENTO FLAT EN VENTA       Lima, Lima, Santiago de Surco   
8          TERRENO URBANO EN VENTA               Lima, Lima, La Molina   
9       DEPARTAMENTO FLAT EN VENTA  Lima, Lima, San Juan de Lurigancho   

               Area Construida                   Area Total       Pisos  \
0   Área Construida : 70.26 m²      Área Ocupada : 70.26 m²  Pisos : 13   
1      Área Terreno : 90.00 m2   Área Construida : 30.00 m²   Pisos : 2   
2  Área Construida : 144.00 m²     Área Ocupada : 144.00 m²   Pisos : 4   
3     Área Terreno : 111.70 m2  Área Construida : 351.00 m²   Pisos : 3   
4     Área Terreno : 400.00 m2    Área Construida : 0.00 m²   Pisos : 0   
5  Área Construida : 225.00 m²          Área Ocupada : 0 m²   Pisos : 1   
6  Área Construida : 144.00 m²     Área Ocupada : 144.00 m²   Pisos : 4   
7  Área Construida : 150.00 m²     Área Ocupada : 190.00 m²   Pisos : 1   
8     Área Terreno : 122.00 m2  Área Construida : 102.00 m²   Pisos : 1   
9  Área Construida : 144.00 m²     Área Ocupada : 144.00 m²   Pisos : 2   

         dormitorios      baños       Cochera  Precio Soles  Precio Dolares  
0   Habitaciones : 2  Baños : 2  Cocheras : 0          1700             456  
1   Habitaciones : 5  Baños : 2  Cocheras : 0        333900           90000  
2   Habitaciones : 3  Baños : 3  Cocheras : 2          8162            2200  
3  Habitaciones : 10  Baños : 3  Cocheras : 0       1261400          340000  
4   Habitaciones : 0  Baños : 0  Cocheras : 0         51940           14000  
5   Habitaciones : 3  Baños : 2  Cocheras : 0          6000            1612  
6   Habitaciones : 3  Baños : 3  Cocheras : 2       1966300          530000  
7   Habitaciones : 3  Baños : 2  Cocheras : 1        760550          205000  
8   Habitaciones : 0  Baños : 0  Cocheras : 0        481929          129900  
9   Habitaciones : 4  Baños : 1  Cocheras : 0        363580           98000

## Escaneo de informacion (Dimension y Tipo de datos)

In [6]:
df.shape

(14400, 10)

In [7]:
df.dtypes

Tipo de Inmueble    object
Direccion           object
Area Construida     object
Area Total          object
Pisos               object
dormitorios         object
baños               object
Cochera             object
Precio Soles         int64
Precio Dolares       int64
dtype: object

## Normalizado del nombre de las columnas

In [8]:
df.columns = [col.strip().replace(' ', '_').lower() for col in df.columns]

## Limpieza de datos

En este apartado se obtendra los valores netos sean INT o FLOAT de las diferentes columnas

In [9]:
df['precio_soles'] = pd.to_numeric(df['precio_soles'], errors='coerce')
df['precio_dolares'] = pd.to_numeric(df['precio_dolares'], errors='coerce')
df['area_construida'] = df['area_construida'].str.extract('(\d+\.?\d*)').astype(float)
df['area_total'] = df['area_total'].str.extract('(\d+\.?\d*)').astype(float)
df['pisos'] = df['pisos'].str.extract('(\d+)').astype(int)
df['dormitorios'] = df['dormitorios'].str.extract('(\d+)').astype(int)
df['baños'] = df['baños'].str.extract('(\d+)').astype(int)
df['cochera'] = df['cochera'].str.extract('(\d+)').astype(int)

In [10]:
display(df.head(20))
display(df.shape)

tipo_de_inmueble                           direccion  \
0     DEPARTAMENTO FLAT EN ALQUILER              Lima, Lima, San Miguel   
1              CASA URBANA EN VENTA  Lima, Lima, San Juan de Lurigancho   
2   DEPARTAMENTO DUPLEX EN ALQUILER              Lima, Lima, Miraflores   
3              CASA URBANA EN VENTA             Lima, Lima, La Victoria   
4       TERRENO INDUSTRIAL EN VENTA                 Piura, Paita, Colan   
5               ALMACÉN EN ALQUILER                   Lima, Lima, Lince   
6      DEPARTAMENTO DUPLEX EN VENTA              Lima, Lima, Miraflores   
7        DEPARTAMENTO FLAT EN VENTA       Lima, Lima, Santiago de Surco   
8           TERRENO URBANO EN VENTA               Lima, Lima, La Molina   
9        DEPARTAMENTO FLAT EN VENTA  Lima, Lima, San Juan de Lurigancho   
10       DEPARTAMENTO FLAT EN VENTA  Lima, Lima, San Juan de Lurigancho   
11       DEPARTAMENTO FLAT EN VENTA               Lima, Lima, San Borja   
12          TERRENO URBANO EN VENTA                 Piura, Piura, Piura   
13             CASA URBANA EN VENTA               Lima, Lima, La Molina   
14            DEPARTAMENTO EN VENTA              Lima, Lima, Miraflores   
15             CASA URBANA EN VENTA      Lambayeque, Chiclayo, Pimentel   
16          TERRENO URBANO EN VENTA           Arequipa, Islay, Mollendo   
17     MINIDEPARTAMENTO EN ALQUILER                 Piura, Piura, Piura   
18       DEPARTAMENTO FLAT EN VENTA     La Libertad, Trujillo, Trujillo   
19            DEPARTAMENTO EN VENTA                   Lima, Lima, Lince   

    area_construida  area_total  pisos  dormitorios  baños  cochera  \
0             70.26       70.26     13            2      2        0   
1             90.00       30.00      2            5      2        0   
2            144.00      144.00      4            3      3        2   
3            111.70      351.00      3           10      3        0   
4            400.00        0.00      0            0      0        0   
5            225.00        0.00      1            3      2        0   
6            144.00      144.00      4            3      3        2   
7            150.00      190.00      1            3      2        1   
8            122.00      102.00      1            0      0        0   
9            144.00      144.00      2            4      1        0   
10            30.00       30.00      4            1      1        0   
11           119.00      123.00      2            3      2        3   
12           200.00        0.00      0            0      0        0   
13           122.00      102.00      1            3      2        0   
14           104.69      104.69      0            3      3        1   
15           125.00      375.00      3           10      5        0   
16           203.90        0.00      0            0      0        0   
17            56.51       56.51      4            2      2        0   
18            85.00       85.00      3            3      3        1   
19            61.24       61.24      4            2      2        1   

    precio_soles  precio_dolares  
0           1700             456  
1         333900           90000  
2           8162            2200  
3        1261400          340000  
4          51940           14000  
5           6000            1612  
6        1966300          530000  
7         760550          205000  
8         481929          129900  
9         363580           98000  
10        100170           27000  
11        816200          220000  
12        234000           62903  
13        481929          129900  
14        723450          195000  
15        600000          161290  
16        204050           55000  
17           800             215  
18        445200          120000  
19        519400          140000

(14400, 10)

## Filtrado de informacion por Departamento o Casa

El Dataframe creado a partir del archivo csv contiene informacion de de casas,departamentos,terrenos y chacras. En el caso de estudio solo requerimos informacion relacionada a casas y departamentos.

In [11]:
filtro1 = df['tipo_de_inmueble'].str.contains('DEPARTAMENTO|CASA', case=False, na=False)
df_filtrado = df[filtro1]

In [12]:
display(df_filtrado.head(20))
display(df_filtrado.shape)

tipo_de_inmueble                           direccion  \
0     DEPARTAMENTO FLAT EN ALQUILER              Lima, Lima, San Miguel   
1              CASA URBANA EN VENTA  Lima, Lima, San Juan de Lurigancho   
2   DEPARTAMENTO DUPLEX EN ALQUILER              Lima, Lima, Miraflores   
3              CASA URBANA EN VENTA             Lima, Lima, La Victoria   
6      DEPARTAMENTO DUPLEX EN VENTA              Lima, Lima, Miraflores   
7        DEPARTAMENTO FLAT EN VENTA       Lima, Lima, Santiago de Surco   
9        DEPARTAMENTO FLAT EN VENTA  Lima, Lima, San Juan de Lurigancho   
10       DEPARTAMENTO FLAT EN VENTA  Lima, Lima, San Juan de Lurigancho   
11       DEPARTAMENTO FLAT EN VENTA               Lima, Lima, San Borja   
13             CASA URBANA EN VENTA               Lima, Lima, La Molina   
14            DEPARTAMENTO EN VENTA              Lima, Lima, Miraflores   
15             CASA URBANA EN VENTA      Lambayeque, Chiclayo, Pimentel   
17     MINIDEPARTAMENTO EN ALQUILER                 Piura, Piura, Piura   
18       DEPARTAMENTO FLAT EN VENTA     La Libertad, Trujillo, Trujillo   
19            DEPARTAMENTO EN VENTA                   Lima, Lima, Lince   
21                    CASA EN VENTA           Lima, Lima, Independencia   
22       DEPARTAMENTO FLAT EN VENTA               Lima, Lima, Surquillo   
23       DEPARTAMENTO FLAT EN VENTA       Lima, Lima, Santiago de Surco   
24    DEPARTAMENTO FLAT EN ALQUILER    Lima, Lima, San Martin de Porres   
25     DEPARTAMENTO DUPLEX EN VENTA                    Lima, Lima, Lima   

    area_construida  area_total  pisos  dormitorios  baños  cochera  \
0             70.26       70.26     13            2      2        0   
1             90.00       30.00      2            5      2        0   
2            144.00      144.00      4            3      3        2   
3            111.70      351.00      3           10      3        0   
6            144.00      144.00      4            3      3        2   
7            150.00      190.00      1            3      2        1   
9            144.00      144.00      2            4      1        0   
10            30.00       30.00      4            1      1        0   
11           119.00      123.00      2            3      2        3   
13           122.00      102.00      1            3      2        0   
14           104.69      104.69      0            3      3        1   
15           125.00      375.00      3           10      5        0   
17            56.51       56.51      4            2      2        0   
18            85.00       85.00      3            3      3        1   
19            61.24       61.24      4            2      2        1   
21           200.00      300.00      2            0      0        0   
22            62.98       62.98      5            2      1        1   
23           113.00      113.00      4            3      2        1   
24            80.00       80.00      5            3      1        0   
25            78.74      123.58      4            3      1        0   

    precio_soles  precio_dolares  
0           1700             456  
1         333900           90000  
2           8162            2200  
3        1261400          340000  
6        1966300          530000  
7         760550          205000  
9         363580           98000  
10        100170           27000  
11        816200          220000  
13        481929          129900  
14        723450          195000  
15        600000          161290  
17           800             215  
18        445200          120000  
19        519400          140000  
21        575050          155000  
22        352450           95000  
23        649250          175000  
24          1000             268  
25        315350           85000

(8972, 10)

tipo_de_inmueble     object
direccion            object
area_construida     float64
area_total          float64
pisos                 int32
dormitorios           int32
baños                 int32
cochera               int32
precio_soles          int64
precio_dolares        int64
dtype: object

In [14]:
df_filtrado.dtypes

tipo_de_inmueble     object
direccion            object
area_construida     float64
area_total          float64
pisos                 int32
dormitorios           int32
baños                 int32
cochera               int32
precio_soles          int64
precio_dolares        int64
dtype: object

Pasamos de Tener un DataFrame que tenia 14400 a solo 8972 valores

## Creacion de un CSV que contenga la informacion limpia

In [13]:
df_filtrado.to_csv('C:\\Users\\ADMIN\\Desktop\\prueba1\\Remax_filtrado_Venta_Alquiler.csv', index=False)